draft-ietf-mls-protocol.md

```diff
@@ -3106,7 +3106,7 @@ The creator of a group MUST take the following steps to initialize the group:
 * Initialize a one-member group with the following initial values:
   * Ratchet tree: A tree with a single node, a leaf containing an HPKE public
     key and credential for the creator
-  * Group ID: A value set by the creator
+  * Group ID: A fresh random value of size `AEAD.Nk`
```

Comment:
There was never a requirement before that the group id had to have a specific internal structure, format, or length. Why does the protocol suddenly care now? It just needs to be unique enough, and you have specified that in the rest of the diff below. 

s/fresh random value of size `AEAD.Nk`/a unique value set by the creator/

Reply:
I agree that it would be prohibitive to require a special structure for group IDs. This would potentially make it difficult to use MLS in existing environments, where group IDs already exist.

It's also not clear to me why `AEAD.Nk` is a good reference for entropy. It depends on the ciphersuite and is therefore orthogonal to the requirements of uniqueness in a system. Moreover, a system that supports several cipehrsuites would see a conflict here.

Reply:
I would probably use `KDF.Nh`, since that's what we use for other unique values (e.g., PSK nonces).